### ISSUE TITLE
BrowserAnimationsModule not detected even after Import

### ISSUE THREAD

**Issue by TaylorZaneKirk** [angular/angular#18751]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

```
ERROR Error: Found the synthetic property @slideInOut. Please include either "BrowserAnimationsModule" or "NoopAnimationsModule" in your application.
    at checkNoSyntheticProp (platform-browser.es5.js:2930)
    at EmulatedEncapsulationDomRenderer2.webpackJsonp.../../../platform-browser/@angular/platform-browser.es5.js.DefaultDomRenderer2.setProperty (platform-browser.es5.js:2898)
    at BaseAnimationRenderer.webpackJsonp.../../../platform-browser/@angular/platform-browser/animations.es5.js.BaseAnimationRenderer.setProperty (animations.es5.js:473)
    at DebugRenderer2.webpackJsonp.../../../core/@angular/core.es5.js.DebugRenderer2.setProperty (core.es5.js:13738)
    at setElementProperty (core.es5.js:9386)
    at checkAndUpdateElementValue (core.es5.js:9305)
    at checkAndUpdateElementInline (core.es5.js:9239)
    at checkAndUpdateNodeInline (core.es5.js:12332)
    at checkAndUpdateNode (core.es5.js:12278)
    at debugCheckAndUpdateNode (core.es5.js:13139)
```

## Expected behavior
Should run without error.

## Minimal reproduction of the problem with instructions
1. Imported BrowserAnimationsModule in app.module.ts
```
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
```

2. Added to imports in app.module.ts
```
imports: [
    BrowserModule,
    FormsModule,
    HttpModule,
    routing,
    Ng2SmartTableModule,
    TreeModule,
    BrowserAnimationsModule
  ]
```

3. In my nav-bar component:
```
import { trigger, state, style, transition, animate } from '@angular/animations';
.
.
.
Component({
  selector: 'app-navigation',
  templateUrl: './navigation.component.html',
  styleUrls: ['./navigation.component.css'],
  animations: [
    trigger('slideInOut', [
      state('in', style({
        transform: 'translate3d(0, 0, 0)'
      })),
      state('out', style({
        transform: 'translate3d(100%, 0, 0)'
      })),
      transition('in => out', animate('400ms ease-in-out')),
      transition('out => in', animate('400ms ease-in-out'))
    ])
  ]
})
```
4. In the html of the nav-bar:
```
<button (click)="toggleMenu()" class="hamburger">
    <span>Button Text</span>
</button>
```

## What is the motivation / use case for changing the behavior?

It would be nice to have animations?

## Environment

<pre><code>
Angular version: 4.3.5

Browser:
- [ x] Chrome (desktop) version 59.0.3071.115
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 6.11.1
- Platform:  Windows

Others:

</code></pre>

**Comment by matsko** [comment]
Hello @TaylorZaneKirk. Everything in here does look correct; I'm unable to figure out why it's not working. Any chance you can craft a demo of this together using https://stackblitz.com/? Thank you.

**Comment by TaylorZaneKirk** [comment]
@matsko 
https://angular-1tgmjh.stackblitz.io

I am unable to reproduce the error on my laptop at home. I will check everything on my work machine on Monday.

I think my issue was that my animation logic was inside the component I was trying to animate instead of that component's parent.

**Comment by rrpatilcse** [comment]
I am also facing same error even after importing BrowserAnimationsModule.
![errorsnap](https://user-images.githubusercontent.com/22540354/31276620-7dadbbc0-aaba-11e7-9b9d-ed0aeee9a492.PNG)

**Comment by matsko** [comment]
@TaylorZaneKirk @rrpatilcse can you guys use RC9 and see if the issue still pops up?

**Comment by j0hnskot** [comment]
I was getting the same error. As @TaylorZaneKirk said, the issue wasn't the missing import but the missing declaration of the animation in the component. 
Using RC9

**Comment by ferrun** [comment]
Have the same error, when use animation inside component. 
![image](https://user-images.githubusercontent.com/4489915/32511446-bd1ffbfa-c42e-11e7-86fb-a45248944074.png)
It's template code:
`<character-preview *ngIf="selectedCharacter"
  [selectedCharacter]="selectedCharacter"
  [@fadeAnimation]
></character-preview>`

**Comment by Czaste** [comment]
Also same issue, working on it for a week now with no success. Making every possible 'change' based on others past experience.

We need something to be done about this, for as stated above, 'It would be nice to have animations?'

_ERROR Error: Found the synthetic property @showState. Please include either "BrowserAnimationsModule" or "NoopAnimationsModule" in your application.
    at checkNoSyntheticProp (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:29339)
    at EmulatedEncapsulationDomRenderer2.DefaultDomRenderer2.setProperty (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:29307)
    at DebugRenderer2.setProperty (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:25047)
    at setElementProperty (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:20708)
    at checkAndUpdateElementValue (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:20627)
    at checkAndUpdateElementInline (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:20561)
    at checkAndUpdateNodeInline (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:23644)
    at checkAndUpdateNode (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:23591)
    at debugCheckAndUpdateNode (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:24448)
    at debugCheckRenderNodeFn (vendor.js?v=ImC4O8Qqs6LILqRyEBT5MpneFa6D4y5w70aVG5MhWrY:sourcemap:24427)_

**Comment by gatimus** [comment]
I am also experiencing this, working on duplication repo 
Could be related to https://github.com/angular/angular/issues/19921

**Comment by j0hnskot** [comment]
@Czaste  Are you declaring the "showState" trigger on the component's animations? I can only see the declaration of "logoAni" on your code.

**Comment by gatimus** [comment]
Never mind, I had a directive with a HostListener for '@myAnimation.done' inside a component with out animations metadata for 'myAnimation' trigger. The error message could be better though.

**Comment by gatimus** [comment]
@Czaste It looks like something in `home.component.htm` is binded to `@showState` but `home.component.ts` only has `@logoAni`

**Comment by Czaste** [comment]
Apologies -- I've been playing with this for several days, and in addition to trying to make my desired animation work, I've been trying very simplistic animations as well (alas all with the same outcome).

Let me get everything together and provide each file so that my issue is clear (and not seemingly a simple error or mistake).

I'll get back to you. Thanks!

**Comment by giscafer** [comment]
same error in my ionic app

angular v5.0.0
 ionic-angular3.9.2

**Comment by CNaiko** [comment]
I am also experiencing this, any progress?

**Comment by dman777** [comment]
:+1:  dealing with this also

**Comment by zarnifoulette** [comment]
Same issue here :
![image](https://user-images.githubusercontent.com/16413502/34420651-0943c5c6-ec0b-11e7-8912-6ca7ee4ea269.png)

**Comment by alok-srivastav** [comment]
Experiencing the same issue. Any solution?

**Comment by IagoBarboza** [comment]
Same issue here! Any solution?

**Comment by Ks89** [comment]
I created a new angular-cli project + scss.
I copied and pasted an example from https://embed.plnkr.co/?show=preview but I'm having the same  problem with `ng serve`

```
ng:///AppModule/AppComponent.ngfactory.js:9 ERROR Error: Found the synthetic property @flyInOut. Please include either "BrowserAnimationsModule" or "NoopAnimationsModule" in your application.
    at checkNoSyntheticProp (webpack-internal:///../../../platform-browser/esm5/platform-browser.js:3057)
    at EmulatedEncapsulationDomRenderer2.DefaultDomRenderer2.setProperty (webpack-internal:///../../../platform-browser/esm5/platform-browser.js:3014)
    at BaseAnimationRenderer.setProperty (webpack-internal:///../../../platform-browser/esm5/animations.js:643)
    at DebugRenderer2.setProperty (webpack-internal:///../../../core/esm5/core.js:15650)
    at setElementProperty (webpack-internal:///../../../core/esm5/core.js:10975)
    at checkAndUpdateElementValue (webpack-internal:///../../../core/esm5/core.js:10894)
    at checkAndUpdateElementInline (webpack-internal:///../../../core/esm5/core.js:10828)
    at checkAndUpdateNodeInline (webpack-internal:///../../../core/esm5/core.js:14120)
    at checkAndUpdateNode (webpack-internal:///../../../core/esm5/core.js:14067)
    at debugCheckAndUpdateNode (webpack-internal:///../../../core/esm5/core.js:14960)
```

**Comment by Ks89** [comment]
Fixed, I made some mistake, because I started again from an empty project and now it's working.

**Comment by olivermuc** [comment]
What was the mistake?

I too see issues, but with a slightly different setup with <parent> being a custom component and driving the child's ***ngIf**:

```
<parent (swipe)="swipeHandler()">
   <div *ngIf="isActive === child1" [@animSlide]="animationDirection">1</div>
   <div *ngIf="isActive === child2" [@animSlide]="animationDirection">2</div>
   <div *ngIf="isActive === child3" [@animSlide]="animationDirection">3</div>
</parent>
```

The above throws the dreaded 
`Please include either "BrowserAnimationsModule" or "NoopAnimationsModule...`
Error.

However if I replace the child divs with custom child components and use `@HostBinding('@animSlide') public animSlide = 'toLeft';` there is no error. But this creates another issue: **I can't dynamically assign an animation direction.**

Any thoughts?

**Comment by olivermuc** [comment]
Here is the solution:

```
<parent (swipe)="swipeHandler()" #parent>
	<child *ngIf="isActive === child1" [animDirection]="parent._animateDir">
			Some content.
	</child>
```

**#parent** needed to be set and referenced in its own child comp reference.
Make sure to use setTimeout on changing the `*ngIf="isActive` variable to allow for the child comp to update the input val.

**Comment by maxmarkus** [comment]
@olivermuc Thanks. Just for the reference and to avoid performance loss, setTimeout creates macroTask and does full change detection. Better create microTask instead. Eg by using `Promise.resolve(undefined).then(() => this.isActive = true)`, since the callback will be run in the current loop as microTask but after other main operations finish.

**Comment by olivermuc** [comment]
Thanks, @maxmarkus .
Can you please elaborate on what the performance gain is, and if that also ensures that the child comp's parameter 'animDirection' gets properly set before "this.isActive = true" kicks in?
Please note that the 'isActive' variable is controlled by the parent of the <parent> comp (2 levels up from <child>).

If beneficial, the 'isActive' control could of course be handed of to the <parent> comp directly.

**Comment by himharsh1997** [comment]
Do this:
    import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
also include in import : [ ]

**Comment by matsko** [comment]
Hey guys. Please include a working stackblitz example of this error. Otherwise I can't investigate why it's breaking.

**Comment by ShaileshBhatNP** [comment]
Here also same issue. But If I run in VS Code, there is no error. I'm getting the error if I run in Visual Studio IDE. Tried to solve all above solutions. But didn't work..
![errorinconsole](https://user-images.githubusercontent.com/29448807/35848272-85ea136c-0b43-11e8-8a6f-ec2955348963.png)


Project link In [VS Code](https://drive.google.com/open?id=1ny_KGOAG4z-OzqL_vFQK2S6htOxR5myl)

Project link in [Visual Studio IDE](https://drive.google.com/open?id=1upclsDsNrXI-GKVLqkKJw_IPmyrYuxW_)

**Comment by Booster2ooo** [comment]
Same here, I started from a Visual Studio project template and followed the Angular tutorial about routing. As soon as I tried to use the animations, the error showed up.

I tried to put a demo on stackblitz, but as soon as the animations package is imported in the project, an error regarding zone.js polyfill appears: https://stackblitz.com/edit/angular-wbebdu

**Comment by earthican** [comment]
I'm getting this same error in Angular v5.1.1.
I've included `BrowserAnimationsModule` in my app module and even have `"@angular/animations": "^5.2.7"` in my `package.json`

**Comment by piotrkabacinski** [comment]
The error is a little bit misleading, I was struggling with the same issue until I realized there was an error in animation declaration. When I fixed it there was no exception.

**Comment by Jahedur** [comment]
this is crazy. Import NoopAnimationsModule instead of BrowserAnimationsModule.

import { NoopAnimationsModule } from '@angular/platform-browser/animations';

This solved my problem.

**Comment by beast195** [comment]
This might help

https://stackoverflow.com/questions/43241193/found-the-synthetic-property-panelstate-please-include-either-browseranimati/43261613

**Comment by Booster2ooo** [comment]
To follow-up on my comment, I updated all packages and tools, tried from scratch again and it worked fine. I guess it was either an error on my side or versions conflict. 😅

**Comment by steveblue** [comment]
Im seeing this same issue but only when bundling with Closure Compiler and @angular 5.2.9. ngc in --watch works fine and BrowserAnimationsModule resolves when requesting the UMD module. 

When bundled with Closure Compiler `ERROR Error: Found the synthetic property @@0:register. Please include either "BrowserAnimationsModule" or "NoopAnimationsModule" in your application.`  

I'm importing `BrowserAnimationsModule` into my app.module. I can post an example repo soon if I cant figure out a fix.

**Comment by ghetolay** [comment]
I just had the same error because I was referencing a no more existing trigger on my template. And looks like most problems here are also about mistakes on defining/declaring trigger/states. 

So the issue in those cases is the miss-leading error message.
Here is a [stackblitz](https://stackblitz.com/edit/angular-hmfpep?file=app%2Fapp.component.html) with an error about a missing trigger which throw the error message about modules.

**Comment by haobr** [comment]
Hi all. 
I've inserted: 
`import { appModuleAnimation } from '@shared/animations/routerTransition';`
and 
`@Component({
  selector: 'app-quan-ly-the',
  templateUrl: './quan-ly-the.component.html',
  styleUrls: ['./quan-ly-the.component.css'],
  **animations: [appModuleAnimation()]**
})`
and this is my routerTransition file: 
`import { trigger, state, animate, style, transition } from '@angular/core';

export function appModuleAnimation() {
    return slideFromBottom();
}

export function accountModuleAnimation() {
    return slideFromUp();
}

export function slideFromBottom() {
    return trigger('routerTransition', [
        state('void', style({ 'padding-top': '20px', opacity: '0' })),
        state('*', style({ 'padding-top': '0px', opacity: '1' })),
        transition(':enter', [
            animate('0.33s ease-out', style({ opacity: '1', 'padding-top': '0px' }))
        ])
    ]);
}

export function slideFromUp() {
    return trigger('routerTransition', [
        state('void', style({ 'margin-top': '-10px', opacity: '0' })),
        state('*', style({ 'margin-top': '0px', opacity: '1' })),
        transition(':enter', [
            animate('0.2s ease-out', style({ opacity: '1', 'margin-top': '0px' }))
        ])
    ]);
}`

**Comment by Tynoe** [comment]
Hie
Recurring error in my console  first in VScode . ng serve --o .then in chrome console,
i had to search and tried may way but these two soved the issue 
 
 Please include either “BrowserAnimationsModule” or “NoopAnimationsModule” in your application

You need to import as illustrated 

In your  C:\...\src\app\app.module.ts

import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';        //<-- copy this.

  imports: [
    BrowserModule,
    BrowserAnimationsModule,               //<--copy this.
    LayoutModule,
    MatToolbarModule,
    MatButtonModule,
    MatSidenavModule,
    MatIconModule,
    MatListModule
  ],

note only 👎 if it fails to solve the err then 
in your C:\...\package.json

remove all ^ on angular : "^6.0.3"

    "@angular/animations": "^6.0.3",
    "@angular/blabla": "^x.x.x",
    "jquery": "^3.0.1",                               //<-- note.

after: you change it should look like this

    "@angular/animations": "6.0.3",
    "@angular/blabla": "x.x.x",
     "jquery": "^3.0.1",                              //<-- do not change.

then delete your ->node_modules  folder and run in your terminal 

      npm install                                      //do not worry i will put back all deleted modules.

**Comment by mszpereira** [comment]
I had the same problem, it was happening because i had this ngIf INSIDE the animation with a true value, while the animation was hidden. removing the ngIf solved errors

<div [@modalvisibility]="vrb1">
    <element *ngIf="vrb2"></dl-login-screen>    
</div>

**Comment by sqnkov** [comment]
Usually, the solution is to import:

`import { BrowserAnimationsModule } from "@angular/platform-browser/animations";` 

after `BrowserModule` in **app.module.ts**

**Comment by flufiorau** [comment]
<div class="reports-element-detail" [@detailExpand]="element == expandedElement ? 'expanded' : 'collapsed'">
@Component({
  selector: 'app-arm-users',
  templateUrl: './arm-users.component.html',
  styleUrls: ['./arm-users.component.scss'],
my way for resolving this problem was next:
I added to component after styleUrls, next strings:
 animations: [
    trigger('detailExpand', [
      state('collapsed', style({height: '0px', minHeight: '0', visibility: 'hidden'})),
      state('expanded', style({height: '*', visibility: 'visible'})),
      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)'))
    ])
  ]

**Comment by kultigin** [comment]
> Here goes another one:
> 
> Setting `display: block;` on the child component's style worked for me. If I remember right, components do not have a default `display` value.

It doesn't do the trick. display:block is necessary to be able to do certain animations, especially transform as I believe but it has nothing to do with this error. My other animations are working fine clearly I have all the right imports for them to work ok. I believe the issue is in *ngIf we use it quite often too... Still I don't have an answer.

**Comment by moradiyainfo** [comment]
YOu have to Chbae Order of imporitn gmodule in app.module.ts  
it will work

**Comment by laguiz** [comment]
I got the same error while trying Angular Material Table with expanded elements. For me it was just missing `animations` declaration on my component. 
https://stackblitz.com/angular/lnyxjmgyxvg?file=app%2Ftable-expandable-rows-example.ts

**Comment by FilippoZerilli** [comment]
Hi guys I fixed this Issue,
in your componenet you have to declare the animations.
Like this:

`animations: [
    trigger('dialog', [
      transition('void => *', [
        style({ transform: 'scale3d(.3, .3, .3)' }),
        animate(100)
      ]),
      transition('* => void', [
        animate(100, style({ transform: 'scale3d(.0, .0, .0)' }))
      ])
    ])
  ]`

if don't declare animations , don't use anything of the import so the component doesn't import the module. At the same time you call this properties with a tag, something like @dialog in the DOM and here is the problem.
Regards,
Filippo

**Comment by albertzubkowicz** [comment]
Same issue here, but application itself compiled just alright. I've encountered mentioned error while running ng test. I've resolved it by adding BrowserAnimationsModule to TestBed imports in your target component.

**Comment by tmjeee** [comment]
Seeing this same error as of Angular 7 as well. Reproduceable using steps given.

**Comment by vitordm** [comment]
I faced the same issue, and I turn it out just put the "BrowserAnimationsModule" inside the "imports" of app.module and inside of component, on "animations",  i added

`@Component({
	selector: 'my-component',
	templateUrl: './my-component.component.html',
	styleUrls: ['./my-component.component.scss'],
	animations: [
		collapseAnimation() <--- Here
	]
})
export class MyComponent ...`

In my case i was using @collapse attribute.

**Comment by erik777** [comment]
I'm having issue, but only with spec.ts.  Otherwise, compiles and runs fine in dev and production.  

Very sad to see this hasn't been resolved in years of being here.  This is what makes Angular a double-edged sword.  You feel very productive, until you run into an issue like this that just becomes a sad time sink.  Lighter is better.

**Comment by lee-merrill** [comment]
Having the same problem here, with only spec.ts, after upgrading to Angular 8 and changing to use the "materials" library. Whether BrowserAnimationsModule is imported or not in app.module.ts doesn't seem to matter, I still get "NullInjectorError: No provider for AnimationBuilder!" when I run "ng test" and it tries to test just loading the application. But the application runs just fine via "ng serve".

**Comment by lenichols** [comment]
@lee-merrill + 1 
- same here... only happens in my test after upgrading to Angular 8.2

**Comment by kleberksms** [comment]
component:

Verify if you load RouterOutlet and add animation
```
import { Component, OnInit } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { routerTransition } from './../../../animations/animations';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.sass'],
  animations: [routerTransition]
})
export class HomeComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

  getState(outlet: RouterOutlet) {
    return outlet.activatedRouteData.state;
  }
}
```

**Comment by ankit-amar-singh** [comment]
![image](https://user-images.githubusercontent.com/13784453/109298321-65300380-7859-11eb-96eb-ec20fcdd21b8.png)
Facing issue in UTC with jest

**Comment by nkurdydyk** [comment]
I received this error in spec, adding import for NoopAnimationsModule to spec fix the issue
`beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        NoopAnimationsModule,
        ...
        ],
   ...
}));
`

**Comment by AndrewKushnir** [comment]
Closing this ticket in favor of #15581 (the error message would be improved in #41356).
